api/clients/eigenda_client_v2.go

```diff
@@ -0,0 +1,142 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/hashicorp/go-multierror"
+	"math/rand"
+)
+
+// EigenDAClientV2 provides the ability to get blobs from the relay subsystem, and to send new blobs to the disperser.
+type EigenDAClientV2 interface {
+	// GetBlob iteratively attempts to retrieve a given blob with key blobKey from the relays listed in the blobCertificate.
+	GetBlob(ctx context.Context, blobKey corev2.BlobKey, blobCertificate corev2.BlobCertificate) ([]byte, error)
+	GetCodec() codecs.BlobCodec
+	Close() error
```

Comment:
Why no DisperseBlob? This feels like just a RelayClient written this way.
The point of the EigenDAClient (at least the way we have it for v1) is to submit blobs with guarantees. So it:
1. disperses blob to disperser (this just puts the blob in the disperser's queue)
2. then pools the BlobStatus endpoint until the blob is processed
3. returns the blobInfo

Reply:
PUT functionality is forthcoming. I've got it drafted locally, but wanted to get some feedback and limit PR size

Reply:
gotcha my bad!